ARrehab/ARrehab/ViewController.swift

```diff
@@ -11,58 +11,247 @@ import RealityKit
 import ARKit
 import Combine
 
-class ViewController: UIViewController, ARSessionDelegate {
+class ViewController: UIViewController {
     
     @IBOutlet var arView: ARView!
     
-    var hasMapped: Bool!
-
-    let cameraEntity = Player(target: .camera)
+    var visualizedPlanes : [ARPlaneAnchor] = []
+    var activeButtons : [UIButton] = []
+    var trackedRaycasts : [ARTrackedRaycast?] = []
+    
+    var boardState : BoardState = .notMapped
+    
+    
+    var tileGrid: TileGrid?
+    var gameBoard: GameBoard?
+    
+    let playerEntity = Player(target: .camera)
     
     override func viewDidLoad() {
+        
         super.viewDidLoad()
         
-        hasMapped = false
+        //Assign the ViewController class to act as the session's delegate (extension below)
+        arView.session.delegate = self
+        startTracking()
         
-        arView.scene.addAnchor(cameraEntity)
-
+        //Set up the player entity
```

Comment:
Is there going to be logic to progress from board layout -> regular game play?

Reply:
Yah, that's the next step, wanted to push this out there first though in case there were any major changes. Also, that process is dependent on how we proceed with content team.

Reply:
Will be done in a future PR.